src/XamlX.IL.Cecil/CecilTypeCache.cs

```diff
@@ -42,6 +36,10 @@ public IXamlType Resolve(CecilTypeResolveContext resolveContext, TypeReference r
             {
                 rv = SecondLayerCache(resolveContext, reference, definition);
             }
+            else if (reference is FunctionPointerType functionPointerType)
+            {
+                rv = new CecilTypeSystem.CecilFunctionPointerType(functionPointerType);
+            }
```

Comment:
I was confused here for a minute, but I guess it makes sense. Function pointers can't be resolved into a type definition, as it's not defined anywhere, technically.

Reply:
Exactly! I choose to add a new type instead of having CecilType handle both reference+definition and reference-only, which would have been a mess.